dask/dataframe/core.py

```diff
@@ -2275,6 +2275,8 @@ def isin(self, values):
     @insert_meta_param_description(pad=12)
     @derived_from(pd.Series)
     def map(self, arg, na_action=None, meta=no_default):
+        if is_series_like(arg) and is_dask_collection(arg):
+            return series_map(self, arg)
```

Comment:
Should `meta` be passed through here?

Reply:
I don't think so. For this operation the output meta is known always since we have meta on both the series that are coming in. (Ex. the output will have the index of the base, and the data type of the map series)

Reply:
Hmm, we should perhaps error if the user provides metadata that doesn't match our expected meta then. I don't like silently ignoring arguments.

The difficult case is `Series[int].map(Series[int])`, since the missing values may cast to float..

Reply:
For matching metadata: is there a method already around for managing this? Taking 2 meta and checking they agree? 

Ah, for that difficult case is there a planned strategy? I have noticed the case that this happens already in dask where meta is not strictly enforced on computation, especially with the null ints becoming floats. Is there something we should do here to capture that, or just match behavior of other methods that return their "usual intended" (eg. in this case, it's expected to be int, but may come out otherwise in execution).